apps/jobs/views.py

```diff
@@ -2073,3 +2076,157 @@ def get_github_badge_data(
+            if request.user.email not in challenge_phase.allowed_email_ids:
+                response_data = {
+                    "error": "Sorry, you are not allowed to participate in this challenge phase"
+                }
+                return Response(
+                    response_data, status=status.HTTP_403_FORBIDDEN
+                )
+
+    participant_team_id = get_participant_team_id_of_user_for_a_challenge(
+        request.user, challenge_pk
+    )
+    try:
+        participant_team = ParticipantTeam.objects.get(pk=participant_team_id)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {"error": "You haven't participated in the challenge"}
+        return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    all_participants_email = participant_team.get_all_participants_email()
+    for participant_email in all_participants_email:
+        if participant_email in challenge.banned_email_ids:
+            message = "You're a part of {} team and it has been banned from this challenge. \
+            Please contact the challenge host.".format(
+                participant_team.team_name
+            )
+            response_data = {"error": message}
+            return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    # Fetch the number of submissions under progress.
+    submissions_in_progress_status = [
+        Submission.SUBMITTED,
+        Submission.SUBMITTING,
+        Submission.RUNNING,
+    ]
+    submissions_in_progress = Submission.objects.filter(
+        participant_team=participant_team_id,
+        challenge_phase=challenge_phase,
+        status__in=submissions_in_progress_status,
+    ).count()
+
+    if (
+        submissions_in_progress
+        >= challenge_phase.max_concurrent_submissions_allowed
+    ):
+        message = "You have {} submissions that are being processed. \
+                   Please wait for them to finish and then try again."
+        response_data = {"error": message.format(submissions_in_progress)}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    serializer = SubmissionSerializer(
+        data=request.data,
+        context={
+            "participant_team": participant_team,
+            "challenge_phase": challenge_phase,
+            "request": request,
+        },
+    )
+
+    submission_message = {
+        "challenge_pk": challenge_pk,
+        "phase_pk": challenge_phase_pk,
+    }
+
+    if serializer.is_valid():
+        serializer.save()
+        submission = serializer.instance
+
+        submission_message["submission_pk"] = submission.pk
+
+        file_ext = os.path.splitext(request.data["file_name"])[-1]
+        file_name = "submission_files/presigned_url_files/submission_{}/{}{}".format(
+            submission.pk, uuid.uuid4(), file_ext
+        )
+        file_key = file_name
+
+        presigned_url = get_presigned_url_for_file_upload(file_name, file_key)
+        if presigned_url == "":
```

Comment:
same here, either handle exception in the api or send a error response dict with error flag from method